### PR TITLE
[shopsys] frontend API testing on b2b domain

### PR DESCRIPTION
**Pull request by vitek-rostislav**
#### Description, the reason for the PR
We needed a solution for testing other than the 1st domain to be able to test the B2B-domain-specific behavior. This PR solves this with two new test cases:
- `GraphQlB2bDomainTestCase`
- `GraphQlB2bDomainWithLoginTestCase`

There are also added tests for the B2B-domain-specific behavior, namely for the following FE API endpoints:
- `Register` mutation
- `ChangePersonalData` mutation
- `AddNewCustomerUser` mutation
- `EditCustomerUserPersonalData` mutation
- `RemoveCustomerUser` mutation
- `CustomerUsersQuery` query

#### Fixes issues
... 

#### Check the appropriate checkboxes below, please

- [ ] [BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/) 
- [x] New feature 
- [x] I have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)

----
:globe_with_meridians: Live Preview:
  - https://rv-tests-second-domain.odin.shopsys.cloud
  - https://cz.rv-tests-second-domain.odin.shopsys.cloud

### CODE REVIEW

File: packages/framework/build.xml [hunk 1]
```diff
@@ -1470,6 +1470,34 @@
...
+                    <arg value="--testsuite"/>
+                    <arg value="FrontendApiFunctionalB2b"/>
+                    <arg value="--exclude-group"/>
+                    <arg value="multidomain"/>
+                    <arg value="--configuration"/>
+                    <arg value="${path.root}/phpunit.xml"/>
+                </exec>
+            </else>
+        </if>
+    </target>
```
Comment (grossmannmartin) [inline packages/framework/build.xml]: I'm thinking, maybe we could write it in a way that reduces dry?

like 
```xml
<if>
    <istrue value="${domains-info.is-multidomain}"/>
    <then>
        <property name="excludeGroup" value="singledomain"/>
    </then>
    <else>
        <property name="excludeGroup" value="multidomain"/>
    </else>
</if>
<exec executable="${path.phpunit.executable}" logoutput="true" passthru="true" checkreturn="true">
    <arg value="--colors=always"/>
    <arg value="--testsuite"/>
    <arg value="FrontendApiFunctionalB2b"/>
    <arg value="--exclude-group"/>
    <arg value="${excludeGroup}"/>
    <arg value="--configuration"/>
    <arg value="${path.root}/phpunit.xml"/>
</exec>
```

what do you think?

File: project-base/app/tests/FrontendApiBundle/Functional/Customer/User/RegisterTest.php [hunk 2]
```diff
@@ -133,4 +147,20 @@ public static function getRegisterQueryVariables(
...
         ];
     }
+
+    /**
+     * @param array $response
+     */
+    private function assertSuccessfulRegistration(array $response): void
+    {
+        $graphQlType = 'Register';
+        $this->assertResponseContainsArrayOfDataForGraphQlType($response, $graphQlType);
```
Comment (grossmannmartin) [inline project-base/app/tests/FrontendApiBundle/Functional/Customer/User/RegisterTest.php]: this assertion is not quite necessary. the same does the `getResponseDataForGraphQlType` internally

File: project-base/app/tests/FrontendApiBundle/Test/CommonGraphQlWithLoginTestCase.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+     * @return string
+     */
+    private static function getLoginQuery(
+        ?string $customerUserEmail = null,
+        ?string $customerUserPassword = null,
+    ): string {
+        $customerUserEmail = $customerUserEmail ?? self::DEFAULT_USER_EMAIL;
+        $customerUserPassword = $customerUserPassword ?? self::DEFAULT_USER_PASSWORD;
+
+        return '
```
Comment (grossmannmartin) [inline project-base/app/tests/FrontendApiBundle/Test/CommonGraphQlWithLoginTestCase.php]: nitpick. we may use `project-base/app/tests/FrontendApiBundle/Functional/Login/graphql/LoginMutation.graphql`

File: project-base/app/tests/FrontendApiBundle/Functional/Customer/User/RegisterTest.php [hunk 4]
```diff
@@ -43,6 +43,17 @@ public function testUniqueBillingAddressIsNotValidatedInB2cRegistration(): void
         $this->assertSuccessfulRegistration($response);
     }
 
+    public function testCompanyUserRegistration(): void
+    {
+        $registerQueryVariables = self::getRegisterQueryVariables();
+        $registerQueryVariables['companyCustomer'] = true;
+        $registerQueryVariables['companyNumber'] = '999';
```
Comment (grossmannmartin) [inline project-base/app/tests/FrontendApiBundle/Functional/Customer/User/RegisterTest.php:50]: this company number will pass the validation? I thought it has to be 8 digits

**Review by grossmannmartin — CHANGES_REQUESTED** [review]

**Review by grossmannmartin — APPROVED** [review]

File: project-base/app/tests/FrontendApiBundle/Functional/Customer/User/RegisterTest.php (hunk 4 above)
Comment (vitek-rostislav) [reply project-base/app/tests/FrontendApiBundle/Functional/Customer/User/RegisterTest.php:50]: seems like there is no such validation on the API side

https://github.com/shopsys/shopsys/blob/16.0/packages/frontend-api/src/Resources/config/graphql-types/ModelType/Customer/BillingAddress/Input/CompanyInputObjectDecorator.types.yaml#L23

File: project-base/app/tests/FrontendApiBundle/Test/CommonGraphQlWithLoginTestCase.php (hunk 3 above)
Comment (vitek-rostislav) [reply project-base/app/tests/FrontendApiBundle/Test/CommonGraphQlWithLoginTestCase.php]: good point, fixed in 120747299f

File: project-base/app/tests/FrontendApiBundle/Functional/Customer/User/RegisterTest.php (hunk 2 above)
Comment (vitek-rostislav) [reply project-base/app/tests/FrontendApiBundle/Functional/Customer/User/RegisterTest.php]: :+1: 94bd1a112a

File: packages/framework/build.xml (hunk 1 above)
Comment (vitek-rostislav) [reply packages/framework/build.xml]: nice, what about this? 0b5eba0984 I was thinking about removing the `domains-info.is-multidomain` completely as it was used just for this purpose, however, it feels quite general and removing could be an unnecessary BC break (however, YAGNI? :shrug: :smile: )

Comment (vitek-rostislav) [reply packages/framework/build.xml]: nice, what about this? 0b5eba0984 I was thinking about removing the `domains-info.is-multidomain` completely as it was used just for this purpose, however, it feels quite general and removing could be an unnecessary BC break (however, YAGNI? :shrug: :smile: )

**Pull request merged**